AODVv2-pandoc-edits/AODVv2.mkd

```diff
@@ -411,12 +411,12 @@ Invalid
     is not referred to as a valid route. Invalid routes contain sequence number information which allows
     incoming information to be assessed for freshness.
 
-If the Local Route Set is stored separately from the Routing Information Base, routes are added to the Routing Information Base
-when LocalRoute.State is valid (set to Active or Idle), and removed from the Routing Information Base when LocalRoute.State becomes Invalid.
+If the Local Route Set is stored separately from the Routing Information Base, routes MAY be added to the Routing Information Base when LocalRoute.State is valid (set to Active or Idle), and MUST be removed from the Routing Information Base when LocalRoute.State becomes Invalid. 
+
+Multiple valid routes for the same address and prefix length but for different metric types may exist in the Local Route Set, but only one route to a particular address and prefix length may exist in the Routing Information Base. The decision of which of the routes should be installed in the Routing Information Base is outside the scope of AODVv2. A LocalRoute should be marked if installed in the Routing Information Base, so that when the LocalRoute changes, the entry in the Routing Information Base can also be updated. 
```

Comment:
I think it would be better to allow multiple routes to be stored in the 
RIB, differentiated by metric type.

If there is only one, then the host is prevented from running 
applications that need different metric types for their preferred choice 
of route.

Regards,
Charlie P.

PS. Any help for me to solve my "Detached Head" problem?



On 6/29/2016 4:55 AM, pv0 wrote:
>
> In AODVv2-pandoc-edits/AODVv2.mkd 
> <https://github.com/Lotterleben/AODVv2-Draft/pull/25#discussion_r68932890>:
>
> > @@ -411,12 +411,12 @@ Invalid
> >      is not referred to as a valid route. Invalid routes contain sequence number information which allows
> >      incoming information to be assessed for freshness.
> >
> > -If the Local Route Set is stored separately from the Routing Information Base, routes are added to the Routing Information Base
> > -when LocalRoute.State is valid (set to Active or Idle), and removed from the Routing Information Base when LocalRoute.State becomes Invalid.
> > +If the Local Route Set is stored separately from the Routing Information Base, routes MAY be added to the Routing Information Base when LocalRoute.State is valid (set to Active or Idle), and MUST be removed from the Routing Information Base when LocalRoute.State becomes Invalid.
> > +
> > +Multiple valid routes for the same address and prefix length but for different metric types may exist in the Local Route Set, but only one route to a particular address and prefix length may exist in the Routing Information Base. The decision of which of the routes should be installed in the Routing Information Base is outside the scope of AODVv2. A LocalRoute should be marked if installed in the Routing Information Base, so that when the LocalRoute changes, the entry in the Routing Information Base can also be updated.
>
> It's a guideline for the "out of scope" bit, I guess. What about this? 
> "The decision of the metric type for which routes are installed in the 
> Routing Information Base is outside the scope of AODVv2."
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Lotterleben/AODVv2-Draft/pull/25/files/3b81d157a63409bc48c6a56fc11846c35d5cccc2#r68932890>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ABbdvs3osuyGLxVKxoOEPFb71n4u3B7Sks5qQl0agaJpZM4I0x3y>.
>